File: rb/features.rb
# RedTideInformation SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module RedTideInformationFeatures
  def self.make_feature(name)
    case name
    when "base"
      RedTideInformationBaseFeature.new
    when "ratelimit"
      RedTideInformationRatelimitFeature.new
    when "retry"
      RedTideInformationRetryFeature.new
    when "test"
      RedTideInformationTestFeature.new
    when "timeout"
      RedTideInformationTimeoutFeature.new
    else
      RedTideInformationBaseFeature.new
    end
  end
end

File: py/features.py
# RedTideInformation SDK feature factory

from feature.base_feature import RedTideInformationBaseFeature
from feature.test_feature import RedTideInformationTestFeature


def _make_feature(name):
    features = {
        "base": lambda: RedTideInformationBaseFeature(),
        "test": lambda: RedTideInformationTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
